control/sum.go

```diff
@@ -35,24 +35,48 @@ func newSum(config BlockConfig, logger logging.Logger) (Block, error) {
 
 func (b *sum) Next(ctx context.Context, x []*Signal, dt time.Duration) ([]*Signal, bool) {
 	if len(x) != len(b.operation) {
-		return b.y, false
-	}
-	y := 0.0
-	for i := range x {
-		op, ok := b.operation[x[i].name]
-		if !ok {
-			return b.y, false
+		lin := 0.0
+		ang := 0.0
+		for i := range x {
+			op, ok := b.operation[x[i].name]
+			if !ok {
+				return b.y, false
+			}
+			switch op {
+			case addition:
+				if i%2 == 0 {
+					lin += x[i].GetSignalValueAt(0)
+				} else {
+					ang += x[i].GetSignalValueAt(0)
+				}
+			case subtraction:
+				if i%2 == 0 {
+					lin -= x[i].GetSignalValueAt(0)
+				} else {
+					ang -= x[i].GetSignalValueAt(0)
+				}
+			}
```

Comment:
Hmmm, it seems like we could do a bit to make the signal more generally multi-input multi-output, but it might be a follow up ticket. Did you try 
```go
switch 
case len(x) != len(b.operation): 
case len(x) %2 == len(b.operation):
sey y1
default:
set y2
``` 
or some structure like it?

Reply:
yeah I'll take another look at how I can clean this up or generalize it